assets/js/src/mailpoet_url_factory.js

```diff
@@ -0,0 +1,27 @@
+const addReferralId = (url, referralId) => {
+  if (!referralId) {
+    return url;
+  }
+  const refUrl = new URL(url);
+  refUrl.searchParams.set('ref', referralId);
+  return refUrl.toString();
+};
+
+const MailPoetUrlFactory = referralId => ({
+  getFreePlanUrl: (utmCampaign = null, utmMedium = null) => {
+    const url = new URL(addReferralId('https://www.mailpoet.com/free-plan/?utm_source=plugin', referralId));
+    if (utmCampaign) {
+      url.searchParams.set('utm_campaign', utmCampaign);
+    }
+    if (utmMedium) {
```

Comment:
I also don't like too much having two optional positional parameters (you already used them wrong, see below). What about allowing to pass `utmParams` object and have it checked by PropTypes? Or maybe even a new method (`getFreePlanUrlWithUtmParams`)? Or some kind of method to add them to any URL object?